File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        my_dict = {}
        for i in range(len(nums)):
            key = nums[i]
            value = i
            my_dict[key] = value
        
        ans = []

        for i in range(len(nums)):
            key1 = nums[i]
            key2 = target - key1
            if key2 in my_dict and my_dict[key2] != i:
                ans.append(i)
                ans.append(my_dict[key2])  # ✅ fixed this line
                break
        return ans

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& arr, int target) {

        unordered_map<int,int>mp;  // assuming that no duplicate exits
        int n = arr.size();
        vector<int> ans;
        for(int i=0;i<n;i++){
            if(mp.find(target - arr[i]) != mp.end()){
                ans.push_back(i);
                ans.push_back(mp[target - arr[i]]);
            }
            mp[arr[i]] = i;
        }
        return ans;
        
    }
};